notification.c

```diff
@@ -135,6 +135,28 @@ struct mako_notification *get_notification(struct mako_state *state,
 	return NULL;
 }
 
+void close_group_notifications(struct mako_notification *top_notif,
+	       enum mako_notification_close_reason reason) {
+	struct mako_notification *notif, *tmp;
+
+	struct mako_criteria *notif_criteria = create_criteria_from_notification(
+			top_notif, &top_notif->style.group_criteria_spec);
+
+	wl_list_for_each_safe(notif, tmp, &top_notif->state->notifications, link) {
+		if (notif->id == top_notif->id) {
+			continue;
+		}
+		if (!match_criteria(notif_criteria, notif)) {
+			continue;
+		}
+
+		close_notification(notif, reason);
+	}
+
+	free(notif_criteria);
```

Comment:
We should use `destroy_criteria` here instead.

Reply:
Yikes, the existing close_notification uses free directly as well. I bet that's at least one of our outstanding memory leaks. My bad. -_-'

Reply:
Thanks, fixed up this and the other case.